src/fields/fp/mod.rs

```diff
@@ -727,7 +727,16 @@ impl<F: PrimeField> FieldVar<F, F> for FpVar<F> {
             (Constant(s), Constant(d)) => Ok(Constant(*s / *d)),
             (Var(s), Constant(d)) => Ok(Var(s.mul_constant(d.inverse().get()?))),
             (Constant(s), Var(d)) => Ok(Var(d.inverse()?.mul_constant(*s))),
-            (Var(s), Var(d)) => Ok(Var(d.inverse()?.mul(s))),
+            (Var(s), Var(d)) => {
+                let d_inv = Self::new_witness(self.cs(), || {
+                    let d_inv = d.value()?.inverse().ok_or(F::zero());
+                    match d_inv {
+                        Ok(v) => Ok(self.value()? * &v),
+                        Err(v) => Ok(v),
+                    }
```

Comment:
Is the following wrong?
```suggestion
                    let d_inv = d.value()?.inverse().unwrap_or(F::zero());
```